### ISSUE TITLE
Impossible to Search using Triode Spotify Plugin

### ISSUE THREAD

**Issue by nikclayton** [nikclayton/android-squeezer#120]
```
What steps will reproduce the problem?
1. Open the spotify plugin from the Internet Radio section
2. Select "Search"
3. Get an error message : "Spotify Error : bad search params"

What is the expected output? What do you see instead?
Expected output should be a text box with the ability to search in Spotify.
Instead I get an error message : Spotify Error bad search params" and nowhere to input
text.

What version of the product are you using? On what operating system?
Squezer 1.3.0 - Android Kit Kat on a Samsung galaxy S3

Please provide any additional information below.

Let me know if I can help more to solve this.
```

Original issue reported on code.google.com by `aditye.kumar` on 2015-05-24 19:44:27

**Comment by Mark0ne** [comment]
Also having this problem. LG G3, Android 5.0, Squeezer 1.3.0. Would be great to see it resolved.

Pressing search key on top-bar will just search my local library rather than spotify (even if pressed from within the spotify app).

Other features do work ok - for example, search history, Playlists, Top 100 (tracks, artists, albums), What's new etc.

Using the Triode plugin option to make app appear under 'my apps' instead does not work - the link under 'my apps' returns a blank screen. This completely removes the app from the 'internet radio' section.

**Comment by nikclayton** [comment]
https://github.com/nikclayton/android-squeezer/wiki/Apps-support-in-Squeezer

**Comment by cybafriend** [comment]
Still the same Problem. 
Bad param, if you are trying to search spotify.
I am using triode Plugin too...

In the webfrontend oft LMS there you have to fill a field with the search Parameter. This is not possible in squeezer. There you can only hit the search button in Thema section APP ->spotify

Plesse fix this.. This app is very usefull.
Thx

**Comment by elParaguayo** [comment]
I think I've got the same issue. I've got both official and Triode's spotify plugins enabled with only the official one visible. When I click on "Search" --> "New Search" there's no text input, instead I go straight to a screen saying "Search results for {query}" (just like the Deezer issue: https://github.com/nikclayton/android-squeezer/issues/138).

Looking forward to a fix for this.

**Comment by elParaguayo** [comment]
It is possible to search Spotify using the JSON protocol. Presumably the app therefore needs to identify that the "search" entry on the Spotify menu has an "input" requirement. The text can then be passed to the JSON request and proceed from there.

**Comment by elParaguayo** [comment]
I agree with your point about the lack of documentation for the JSON protocol on this. I've been trying to develop my own controller on the raspberry pi using this protocol and it's not quick (and I'm probably not looking at half of the features you are!). Good luck.

**Comment by bramduijsens** [comment]
Just wondering whether any progress is being made transferring to the JSON protocol and solving this issue :) Would be fantastic, because without search spotify is not very functional unfortunately.

**Comment by kaaholst** [comment]
Just to let you know, I am working on implementing the JSON protocol in Squeezer, and while a solution is definitely coming, it is not close.

**Comment by Flautz** [comment]
I added Pull request #184 which fixes that topic.

**Comment by Mark0ne** [comment]
Any chance this could be built into the main stream and released into playstore?  Would love to try it out without having to learn to build APKs.  Cheers

**Comment by nikclayton** [comment]
I've just released this in the first beta of 1.5.0. You can sign up to receive the beta versions at https://github.com/nikclayton/android-squeezer/wiki/Beta-testing.

**Comment by nikclayton** [comment]
1.5.0 is released, which should fix this, so closing this issue. If you still see a problem, please re-open it.
